clippy_lints/src/needless_bool.rs

```diff
@@ -49,11 +50,20 @@ impl LateLintPass for NeedlessBool {
     fn check_expr(&mut self, cx: &LateContext, e: &Expr) {
         use self::Expression::*;
         if let ExprIf(ref pred, ref then_block, Some(ref else_expr)) = e.node {
-            let reduce = |hint: &str, not| {
-                let hint = match snippet_opt(cx, pred.span) {
-                    Some(pred_snip) => format!("`{}{}`", not, pred_snip),
-                    None => hint.into(),
+            let reduce = |ret, not| {
+                let snip = Sugg::hir(cx, pred, "<predicate>");
+                let snip = if not {
+                    !snip
+                } else {
+                    snip
+                };
+
+                let hint = if ret {
+                    format!("return {};", snip)
```

Comment:
The semicolon is wrong here, if the if-then-else is a match arm (or a function argument or sth)